src/SAWScript/Crucible/LLVM/X86.hs

```diff
@@ -571,19 +560,36 @@ llvm_verify_x86_common (Some (llvmModule :: LLVMModule x)) path nm globsyms chec
 
       let execFeatures = simpleLoopFixpointFeature ++ psatf
 
-      liftIO $ C.executeCrucible execFeatures initial >>= \case
-        C.FinishedResult{} -> pure ()
+      finalState <- liftIO $ C.executeCrucible execFeatures initial >>= \case
+        C.FinishedResult _ pr -> do
+          gp <- getGlobalPair opts pr
+          mem' <- maybe
+            (fail "internal error: LLVM Memory global not found")
+            return
+            (C.lookupGlobal mvar $ gp ^. C.gpGlobals)
```

Comment:
There are at least two calls to `lookupGlobal mvar (gp ^. gpGlobals)` in this file (there's another on line 578) that in turn validate that it returns a `Just` value. It might be worth factoring this out into a little helper function.